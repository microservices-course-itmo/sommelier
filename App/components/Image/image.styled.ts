import styled from 'styled-components/native'

type ImageContainerProps = {
  width?: number
  height?: number
  borderRadius?: number
}

export const ImageContainer = styled.Image<ImageContainerProps>`
  width: ${({ width }) => width || 0}px;
  height: ${({ height }) => height || 0}px;
  border-radius: ${({ borderRadius }) => borderRadius || 0}px;
`
