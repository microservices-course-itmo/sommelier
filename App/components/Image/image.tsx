import React from 'react'
import * as Styled from './image.styled'
import { BoxContainer } from '../Box/box.styled'
import theme from '../../theme'

export type Props = {
  image: string
  width?: number
  height?: number
  border?: string
  borderRadius?: number
}

const Image = ({
  image = '',
  width,
  height,
  border,
  borderRadius: radius,
}: Props) => {
  let borderRadius = radius
  if (!borderRadius) {
    borderRadius = Number(border?.split(' ')[0])
  }
  const borderColor = border?.split(' ')[2] as keyof typeof theme.colors
  return (
    <BoxContainer borderRadius={borderRadius} borderColor={borderColor}>
      <Styled.ImageContainer
        source={{ uri: image }}
        width={width}
        height={height}
      />
    </BoxContainer>
  )
}

export default Image
