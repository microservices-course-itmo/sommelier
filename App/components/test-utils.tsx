import React from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const themeRender = (component: any) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
