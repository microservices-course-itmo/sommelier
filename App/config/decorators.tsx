import { ThemeProvider } from 'styled-components'
import React from 'react'
import styled from 'styled-components/native'
import { theme } from '../theme'

const StyledBufferView = styled.View`
  flex: 1;
  margin-vertical: 40;
  margin-horizontal: 20;
`

export const ThemeDecorator = (storyFn: any) => (
  <ThemeProvider theme={theme}>{storyFn()}</ThemeProvider>
)

export const BufferView = (storyFn: any) => (
  <StyledBufferView>{storyFn()}</StyledBufferView>
)
