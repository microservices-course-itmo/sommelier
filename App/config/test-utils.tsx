import React, { ReactNode } from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { theme } from '../theme'

const themeRender = (component: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
