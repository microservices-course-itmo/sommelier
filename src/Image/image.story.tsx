import React from 'react';
import { number, select } from '@storybook/addon-knobs';
import {  } from '@storybook/addon-actions'
import MyImage from './image';
import { border } from './enums/border';
import styled from 'styled-components';

const StyledWrapper = styled.div`
    display: flex; 
`;

export default {
  title: 'Image',
};

export const imageKnob = () => {
  <StyledWrapper>
    <MyImage
      width={number('width', 256)}
      height={number('height', 256)}
      border={select('border', border, border.SOLID)}
      borderRadius={number('borderRadius', 0)}
    ></MyImage>
  </StyledWrapper>
}