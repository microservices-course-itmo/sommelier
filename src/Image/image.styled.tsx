import styled from 'styled-components';

export const StyledImage = styled.image`
  margin: 5%;
  border-radius: 5px;
`;