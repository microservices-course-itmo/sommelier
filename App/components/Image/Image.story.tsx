import React from 'react'
import { storiesOf } from '@storybook/react-native/dist'
import { action } from '@storybook/addon-actions'
import { withKnobs, number, text } from '@storybook/addon-knobs'

import Image from './image'
import { BufferView, ThemeDecorator } from '../../config/decorators'

const defaultImage = () => (
  <Image
    image={text(
      'image',
      'https://i.pinimg.com/originals/53/54/81/535481ed597409956a8e1288987a4ae3.jpg'
    )}
    width={number('width', 128)}
    height={number('height', 128)}
  >
    {text('text', 'this is an image')}
  </Image>
)

storiesOf('Image', module)
  .addDecorator(BufferView)
  .addDecorator(ThemeDecorator)
  .addDecorator(withKnobs)
  .add('Default image', defaultImage)
