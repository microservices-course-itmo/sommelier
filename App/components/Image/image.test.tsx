import React from 'react'
import { themeRender } from '../../config/test-utils'
import Image from './image'

describe('Image component', () => {
  it('Component matches snapshot', () => {
    const tree = themeRender(
      <Image image='https://i.pinimg.com/originals/53/54/81/535481ed597409956a8e1288987a4ae3.jpg' />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Width prop assigns correctly', () => {
    const elem = themeRender(
      <Image
        image='https://i.pinimg.com/originals/53/54/81/535481ed597409956a8e1288987a4ae3.jpg'
        width={50}
      />
    )
    expect(elem.toJSON().children[0].props.width).toBe(50)
  })
  it('Height prop assigns correctly', () => {
    const elem = themeRender(
      <Image
        image='https://i.pinimg.com/originals/53/54/81/535481ed597409956a8e1288987a4ae3.jpg'
        height={50}
      />
    )
    expect(elem.toJSON().children[0].props.height).toBe(50)
  })
  it('Border prop assigns correctly', () => {
    const elem = themeRender(
      <Image
        image='https://i.pinimg.com/originals/53/54/81/535481ed597409956a8e1288987a4ae3.jpg'
        border='solid'
      />
    )
    expect(elem.toJSON().children[0].props.border).toBe('solid')
  })
  it('Border radius prop assigns correctly', () => {
    const elem = themeRender(
      <Image
        image='https://i.pinimg.com/originals/53/54/81/535481ed597409956a8e1288987a4ae3.jpg'
        borderRadius={5}
      />
    )
    expect(elem.toJSON().props.borderRadius).toBe(5)
  })
})
