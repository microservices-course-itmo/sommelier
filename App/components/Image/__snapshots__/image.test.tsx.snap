// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component Component matches snapshot 1`] = `
<View
  borderRadius={NaN}
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "borderColor": "#000",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.pinimg.com/originals/53/54/81/535481ed597409956a8e1288987a4ae3.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 0,
          "height": 0,
          "width": 0,
        },
      ]
    }
  />
</View>
`;
